models/Comment.php

```diff
@@ -66,60 +62,63 @@ function get_comments(){
 
     // Fonction permettant de supprimer un commentaire
     function delete_comment(){
-        if (isset($_GET['delete']) AND !empty($_GET['delete'])) {
+        $delete = filter_input(INPUT_GET, 'delete', FILTER_SANITIZE_NUMBER_INT);
         
-        $delete = (int) $_GET['delete'];
-
-        $db = getPdo();
-        
-        $req = $db->prepare('DELETE FROM comments WHERE id = ?');
+        $req = $this->db->prepare('DELETE FROM comments WHERE id = ?');
         $req->execute(array($delete));
 
-        header("Location: index.php?page=dashboard");
-        
-        exit();    
-        }
+        header("Location: /dashboard");
     }
 
     // Fonction permettant d'approuver un commentaire
     function approve_comment(){
-        if (empty($_GET['approve']) || !ctype_digit($_GET['approve'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
-        }
-
-        $id = $_GET['approve'];
-
-        $db = getPdo();
+        $id = filter_input(INPUT_GET, 'approve', FILTER_SANITIZE_NUMBER_INT);
 
-        $query = $db->prepare('UPDATE comments SET seen=1 WHERE id=:id');
+        $query = $this->db->prepare('UPDATE comments SET seen=1 WHERE id=:id');
+        
         $query->execute(['id' => $id]);
 
-        header("Location: index.php?page=dashboard");
-
-        exit();
+        header("Location: /dashboard");
     }
 
-    // EN COURS
-    //Fonction permettant de supprimer tous les commentaires d'un article (dans le cas où on veut supprimer un article)
-    function delete_article_comments(){
-        
-        if (empty($_GET['id']) || !ctype_digit($_GET['id'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
+    
+    // Fonction permettant de vérifier que les données envoyées pat l'utilisateur sont bonnes.
+    function form_comment_verification(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method form_comment_verification() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5349276)